NaturalOrderComparator.java

```diff
@@ -41,33 +39,25 @@ int compareRight(String a, String b)
             char ca = charAt(a, ia);
             char cb = charAt(b, ib);
 
-            if (!Character.isDigit(ca) && !Character.isDigit(cb))
-            {
+            if (!Character.isDigit(ca) && !Character.isDigit(cb)) {
                 return bias;
             }
-            if (!Character.isDigit(ca))
-            {
+            if (!Character.isDigit(ca)) {
                 return -1;
             }
-            if (!Character.isDigit(cb))
-            {
+            if (!Character.isDigit(cb)) {
                 return +1;
             }
-            if (ca < cb)
-            {
-                if (bias == 0)
-                {
+            if (ca == 0 && cb == 0) {
+                return bias;
+            }
+
+            if (ca < cb) {
```

Comment:
You could pull the `bias == 0` condition out, and nest the `ca > cb` and `ca < cb` inside it.